builder/job.go

```diff
@@ -236,12 +231,42 @@ func Build(d *daemon.Daemon, buildConfig *Config) error {
 	if err != nil {
 		return err
 	}
-	if repoName != "" {
-		return d.Repositories().Tag(repoName, tag, id, true)
+
+	for _, rt := range repoAndTags {
+		if err := d.Repositories().Tag(rt.repo, rt.tag, id, true); err != nil {
+			return err
+		}
 	}
 	return nil
 }
 
+// parseRepoAndTags parses the raw "t" argument received from the client
+// to a slice of repoAndTag.
+// It also validates each repoName and tag.
+func parseRepoAndTags(names []string) ([]repoAndTag, error) {
+	repoAndTags := []repoAndTag{}
+	// The "t" parameter is optional.
+	if names == nil {
+		return repoAndTags, nil
```

Comment:
no need to allocate an empty slice, you can just `return nil, nil` and define `var repoAndTags []repoAndTag` after that. `append` will allocate whatever it needs even if the slice is nil.